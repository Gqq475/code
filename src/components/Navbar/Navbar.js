import React from 'react'
import { connect } from 'react-redux'
import classes from './Navbar.scss'
import { Link } from 'react-router'
import { Menu, Popover, Position, Button, MenuItem } from '@blueprintjs/core'
import SubNav from 'components/SubNav'
import { deauthenticate } from 'actions'
import ReLogin from 'components/ReLogin'
import { identifyRole } from '../../Utils.js'

type Props = {
  route: Object,
  lockScreen: Boolean,
  userRole: Array,
  deauthenticate: Function
}

export class Navbar extends React.Component {
  props: Props;

  render () {
    let { userRole } = this.props
    let path = null
    let RouteLink = null
    if ((/qca/).exec(this.props.route.path)) {
      path = 'QCA'
    } else if ((/drtm/).exec(this.props.route.path)) {
      path = 'dRTM'
    } else {
      path = 'SADJ'
    }
    if ((/QCA/).exec(path)) {
      RouteLink = (
        <div style={{
          display: 'flex'
        }}>
          <Link to='/dRTM/currentalarm'
            className={classes.link}>
            {'dRTM'}
          </Link>
          {
            (identifyRole(userRole) !== 'guest-only') ? (
              <Link to={'/ojs/servicemanagement'}
                className={classes.link}>
                {'SADJ'}
              </Link>
            ) : null
          }
        </div>
      )
    } else if ((/SADJ/).exec(path)) {
      RouteLink = (
        <div style={{
          display: 'flex'
        }}>
          <Link to={'/dRTM/currentalarm'}
            className={classes.link}>
            {'dRTM'}
          </Link>
          <Link to={'/qca/dashboard'}
            className={classes.link}>
            {'QCA'}
          </Link>
        </div>
      )
    } else if ((/dRTM/).exec(path)) {
      RouteLink = (
        <div style={{
          display: 'flex'
        }}>
          <Link to={'/qca/dashboard'}
            className={classes.link}>
            {'QCA'}
          </Link>
          {
            (identifyRole(userRole) !== 'guest-only') ? (
              <Link to={'/ojs/servicemanagement'}
                className={classes.link}>
                {'SADJ'}
              </Link>
            ) : null
          }
        </div>
      )
    }
    const dRTMCompassMenu = (
      <Menu className={classes.menu}>
        <Link to='/drtm/currentalarm' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.CURRENTALARM} />
        </Link>
        <Link to='/drtm/alarmhistory' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.ALARMHISTORY} />
        </Link>
        <Link to='/drtm/criteriamanager' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.CRITERIAMANAGER} />
        </Link>

        <Link to='/drtm/alarmrate' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.ALARMRATE} />
        </Link>
      </Menu>
    )
    const QCACompassMenu1 = (
      <Menu className={classes.menu}>
        <Link to='/qca/dashboard' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.DASHBOARD} />
        </Link>
        <Link to='/qca/defectmap' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.DEFECTMAP} />
        </Link>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.膜厚分析} />
        </Link>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={'接触脚功能与破片功能'} />
        </Link>
      </Menu>
    )
    const QCACompassMenu2 = (
      <Menu className={classes.menu}>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={'Format试算功能'} />
        </Link>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.AOI拍照正确性分析} />
        </Link>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.REPAIR即时修补统计} />
        </Link>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.检测机台检出率} />
        </Link><Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.AOI灰阶监控} />
        </Link>
      </Menu>
    )
    const QCACompassMenu3 = (
      <Menu className={classes.menu}>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.AOIMAP对比} />
        </Link>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.PANEL对比分析} />
        </Link>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.REPAIRFORMAT修补功能} />
        </Link>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.TEST即时修补统计} />
        </Link>
      </Menu>
    )
    const QCACompassMenu4 = (
      <Menu className={classes.menu}>
        <MenuItem className={classes.subMenu} text='共通性分析统计功能'>
          <Link to='/qca/stk' className={classes.menuItem}>
            <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
              text={_.搬运路径共通性分析} />
          </Link>
          <Link to='/qca/stk' className={classes.menuItem}>
            <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
              text={'机台共通性分析'} />
          </Link>
          <Link to='/qca/stk' className={classes.menuItem}>
            <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
              text={'原物料共通性分析'} />
          </Link>
          <Link to='/qca/stk' className={classes.menuItem}>
            <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
              text={'制程参数（EDC）比对'} />
          </Link>
          <Link to='/qca/echarts' className={classes.menuItem}>
            <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
              text={'Echarts制图试用'} />
          </Link>
        </MenuItem>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.DEFECTRATE分析} />
        </Link>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.REWORKRATE分析} />
        </Link>
        <Link to='/qca/' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text={_.黄光专区} />
        </Link>
      </Menu>
    )
    const SAJDCompassMenu = (
      <Menu className={classes.menu}>
        <Link to='/ojs/labelingtool' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text='标记工具' />
        </Link>
        <Link to='/ojs/modelversion' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text='模型管理' />
        </Link>
        <Link to='/ojs/modeltrain' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text='模型训练' />
        </Link>
        <Link to='/ojs/modeltest' className={classes.menuItem}>
          <Button className={`pt-menu-item pt-minimal ${classes['menu-item-button']}`}
            text='模型检测' />
        </Link>
      </Menu>
    )
    return (
      <div className={classes['Navbar-container']}>
        {this.props.lockScreen && <ReLogin />}
        <nav className={`pt-navbar pt-dark ${classes.navbar}`}>
          <div className='pt-navbar-group pt-align-left'>
            <div className='pt-navbar-heading'>
              <img className={classes.logo} src='/quality03_0111-20.png' />
            </div>
            <div className={classes.title}>
              {path}
            </div>
            {this.renderList(this.props.userRole, path, SAJDCompassMenu,
             QCACompassMenu4, QCACompassMenu3, QCACompassMenu2, QCACompassMenu1, dRTMCompassMenu)}
          </div>
          <div className='pt-navbar-group pt-align-right'>
            <span className='pt-navbar-divider' />
            { RouteLink }
            <span className='pt-navbar-divider' />
            <button className='pt-button pt-minimal pt-icon-notifications' />
            <button className='pt-button pt-minimal pt-icon-more' />
            <span className='pt-navbar-divider' />
            <span style={{cursor: 'pointer'}} onClick={this.props.deauthenticate}>Log out</span>
          </div>
        </nav>
        <SubNav {...this.props} location={path} />
      </div>
    )
  }

  renderList (userRole, path, SAJDCompassMenu, QCACompassMenu4,
   QCACompassMenu3, QCACompassMenu2, QCACompassMenu1, dRTMCompassMenu) {
    if ((/QCA/).exec(path)) {
      return (
        <section className={classes['navbar-lists']}>
          <Popover className={classes['button-bg']} content={QCACompassMenu1} position={Position.BOTTOM}>
            <span className={classes.button}>
              {_.通用功能}<span className='pt-icon-standard pt-icon-chevron-down pt-align-right' />
            </span>
          </Popover>
          <Popover className={classes['button-bg']} content={QCACompassMenu2} position={Position.BOTTOM}>
            <span className={classes.button}>
              {'检视专区'}<span className='pt-icon-standard pt-icon-chevron-down pt-align-right' />
            </span>
          </Popover>
          <Popover className={classes['button-bg']} content={QCACompassMenu3} position={Position.BOTTOM}>
            <span className={classes.button}>
              {'测试专区'}<span className='pt-icon-standard pt-icon-chevron-down pt-align-right' />
            </span>
          </Popover>
          <Popover className={classes['button-bg']} content={QCACompassMenu4} position={Position.BOTTOM}>
            <span className={classes.button}>
              {'工程专区'}<span className='pt-icon-standard pt-icon-chevron-down pt-align-right' />
            </span>
          </Popover>
        </section>
      )
    } else if ((/dRTM/).exec(path)) {
      return (
        <section className={classes['navbar-lists']}>
          <Popover className={classes['button-bg']} content={dRTMCompassMenu} position={Position.BOTTOM}>
            <span className={classes.button}>
              {_.ALARM}
              <span className='pt-icon-standard pt-icon-chevron-down pt-align-right' />
            </span>
          </Popover>
          <Link to='/drtm/defectjudge' className={`${classes.button} ${classes['button-a']}`}>
            {_.DEFECTJUDGE}
          </Link>
        </section>
      )
    } else if ((/SADJ/).exec(path)) {
      return (
        <section className={classes['navbar-lists']}>
          <Link to='/ojs/servicemanagement' className={`${classes.button} ${classes['button-a']}`}>
            {_.SERVICEMANAGEMENT}
          </Link>
          <Link to='/ojs/autodefectjudge' className={`${classes.button} ${classes['button-a']}`}>
            {_.AUTODEFECTJUDGE}
          </Link>
          {
            (identifyRole(userRole) === 'admin') ? (
              <Popover className={classes['button-bg']} content={SAJDCompassMenu} position={Position.BOTTOM}>
                <span className={classes.button}>
                  Offline
                  <span className='pt-icon-standard pt-icon-chevron-down pt-align-right' />
                </span>
              </Popover>
            ) : null
          }
        </section>
      )
    }
  }
}

export default connect(
  (state) => {
    return {
      token: state.auth.token,
      lockScreen: state.auth.lockScreen,
      userRole: state.auth.userRole
    }
  },
  (dispatch) => ({
    deauthenticate: () => dispatch(deauthenticate())
  })
)(Navbar)
