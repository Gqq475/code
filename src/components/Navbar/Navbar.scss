.Navbar-container {
  .logo {
    height: 35px;
  }

  .navbar {
    background: #0073BB;
    color: #fff;
    padding: 0 80px;
    height: 80px;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .title {
    font-size: 18px;
    font-weight: 500;
    margin: 0 40px 0 20px;
  }

  .link {
    font-size: 18px;
    font-weight: 500;
    width: 80px;
    text-align: center;
    color: #fff;
  }

  .link:hover {
    text-decoration: none;
    color: #fff;
  }

  .button {
    cursor: pointer;
    margin: 0 20px;
    width: 80px;
    color: #fff;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .button-a{
    height: 75px;
    display: inline-block;
    line-height: 80px;
    margin-top: -5px;
  }

  .button:hover {
    text-decoration: none;
    color: #fff;
  }

  .button-bg{
    height: 75px;
    line-height: 80px;
    margin-top: -5px;
  }

  .button-bg:hover,
  .pt-popover-open.button-bg{
    background-image: linear-gradient(to top,#0087DC,#0073BB);
  }
}

.menu {
  background: #0087DC !important;
  padding: 10px 0;

  .menuItem{
    color: #fff;
    box-sizing: border-box;
    padding: 0 4px;
    display: block;

    .menu-item-button{
      text-align: center;
      padding: 10px 0;
    }
  }

  .subMenu {
    margin: 0 4px;
    padding: 10px 0;
    text-align: center;
  }

  .menuItem:hover {
    color: #fff;
    text-decoration: none;

    .menu-item-button{
      background: #0073BB;
    }
  }
}

.navbar-lists {
  display: flex;
  flex-flow: row nowrap;
}
