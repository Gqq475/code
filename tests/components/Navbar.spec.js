import React from 'react'
import { Navbar } from 'components/Navbar/Navbar'

describe('(Component) Navbar', () => {
  it('render normally if path is qca', () => {
    const props = {
      route: {
        path: 'qca(/:path)'
      }
    }
    const wrapper = shallow(
      <Navbar {...props} />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if path is ojs', () => {
    const props = {
      route: {
        path: 'ojs(/:path)'
      }
    }
    const wrapper = shallow(
      <Navbar {...props} />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if path is drtm', () => {
    const props = {
      route: {
        path: 'drtm(/:path)'
      }
    }
    const wrapper = shallow(
      <Navbar {...props} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
