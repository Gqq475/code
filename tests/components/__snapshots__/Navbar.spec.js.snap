// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Navbar render normally if path is drtm 1`] = `
<div
  className="Navbar-container"
>
  <nav
    className="pt-navbar pt-dark navbar"
  >
    <div
      className="pt-navbar-group pt-align-left"
    >
      <div
        className="pt-navbar-heading"
      >
        <img
          className="logo"
          src="/quality03_0111-20.png"
        />
      </div>
      <div
        className="title"
      >
        dRTM
      </div>
      <section
        className="navbar-lists"
      >
        <Popover
          arrowSize={30}
          className="button-bg"
          content={
            <Blueprint.Menu
              className="menu"
            >
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/drtm/currentalarm"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/drtm/alarmhistory"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/drtm/criteriamanager"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/drtm/alarmrate"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
            </Blueprint.Menu>
          }
          defaultIsOpen={false}
          hoverCloseDelay={300}
          hoverOpenDelay={150}
          inheritDarkTheme={true}
          inline={false}
          interactionKind={0}
          isDisabled={false}
          isModal={false}
          popoverClassName=""
          position={7}
          rootElementTag="span"
          transitionDuration={300}
          useSmartArrowPositioning={true}
          useSmartPositioning={false}
        >
          <span
            className="button"
          >
            <span
              className="pt-icon-standard pt-icon-chevron-down pt-align-right"
            />
          </span>
        </Popover>
        <Link
          className="button button-a"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/drtm/defectjudge"
        />
      </section>
    </div>
    <div
      className="pt-navbar-group pt-align-right"
    >
      <span
        className="pt-navbar-divider"
      />
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <Link
          className="link"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/qca/dashboard"
        >
          QCA
        </Link>
      </div>
      <span
        className="pt-navbar-divider"
      />
      <button
        className="pt-button pt-minimal pt-icon-notifications"
      />
      <button
        className="pt-button pt-minimal pt-icon-more"
      />
      <span
        className="pt-navbar-divider"
      />
      <span
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Log out
      </span>
    </div>
  </nav>
  <SubNav
    location="dRTM"
    route={
      Object {
        "path": "drtm(/:path)",
      }
    }
  />
</div>
`;

exports[`(Component) Navbar render normally if path is ojs 1`] = `
<div
  className="Navbar-container"
>
  <nav
    className="pt-navbar pt-dark navbar"
  >
    <div
      className="pt-navbar-group pt-align-left"
    >
      <div
        className="pt-navbar-heading"
      >
        <img
          className="logo"
          src="/quality03_0111-20.png"
        />
      </div>
      <div
        className="title"
      >
        SADJ
      </div>
      <section
        className="navbar-lists"
      >
        <Link
          className="button button-a"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/ojs/servicemanagement"
        />
        <Link
          className="button button-a"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/ojs/autodefectjudge"
        />
      </section>
    </div>
    <div
      className="pt-navbar-group pt-align-right"
    >
      <span
        className="pt-navbar-divider"
      />
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <Link
          className="link"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/dRTM/currentalarm"
        >
          dRTM
        </Link>
        <Link
          className="link"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/qca/dashboard"
        >
          QCA
        </Link>
      </div>
      <span
        className="pt-navbar-divider"
      />
      <button
        className="pt-button pt-minimal pt-icon-notifications"
      />
      <button
        className="pt-button pt-minimal pt-icon-more"
      />
      <span
        className="pt-navbar-divider"
      />
      <span
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Log out
      </span>
    </div>
  </nav>
  <SubNav
    location="SADJ"
    route={
      Object {
        "path": "ojs(/:path)",
      }
    }
  />
</div>
`;

exports[`(Component) Navbar render normally if path is qca 1`] = `
<div
  className="Navbar-container"
>
  <nav
    className="pt-navbar pt-dark navbar"
  >
    <div
      className="pt-navbar-group pt-align-left"
    >
      <div
        className="pt-navbar-heading"
      >
        <img
          className="logo"
          src="/quality03_0111-20.png"
        />
      </div>
      <div
        className="title"
      >
        QCA
      </div>
      <section
        className="navbar-lists"
      >
        <Popover
          arrowSize={30}
          className="button-bg"
          content={
            <Blueprint.Menu
              className="menu"
            >
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/dashboard"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/defectmap"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                  text="接触脚功能与破片功能"
                />
              </Link>
            </Blueprint.Menu>
          }
          defaultIsOpen={false}
          hoverCloseDelay={300}
          hoverOpenDelay={150}
          inheritDarkTheme={true}
          inline={false}
          interactionKind={0}
          isDisabled={false}
          isModal={false}
          popoverClassName=""
          position={7}
          rootElementTag="span"
          transitionDuration={300}
          useSmartArrowPositioning={true}
          useSmartPositioning={false}
        >
          <span
            className="button"
          >
            <span
              className="pt-icon-standard pt-icon-chevron-down pt-align-right"
            />
          </span>
        </Popover>
        <Popover
          arrowSize={30}
          className="button-bg"
          content={
            <Blueprint.Menu
              className="menu"
            >
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                  text="Format试算功能"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
            </Blueprint.Menu>
          }
          defaultIsOpen={false}
          hoverCloseDelay={300}
          hoverOpenDelay={150}
          inheritDarkTheme={true}
          inline={false}
          interactionKind={0}
          isDisabled={false}
          isModal={false}
          popoverClassName=""
          position={7}
          rootElementTag="span"
          transitionDuration={300}
          useSmartArrowPositioning={true}
          useSmartPositioning={false}
        >
          <span
            className="button"
          >
            检视专区
            <span
              className="pt-icon-standard pt-icon-chevron-down pt-align-right"
            />
          </span>
        </Popover>
        <Popover
          arrowSize={30}
          className="button-bg"
          content={
            <Blueprint.Menu
              className="menu"
            >
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
            </Blueprint.Menu>
          }
          defaultIsOpen={false}
          hoverCloseDelay={300}
          hoverOpenDelay={150}
          inheritDarkTheme={true}
          inline={false}
          interactionKind={0}
          isDisabled={false}
          isModal={false}
          popoverClassName=""
          position={7}
          rootElementTag="span"
          transitionDuration={300}
          useSmartArrowPositioning={true}
          useSmartPositioning={false}
        >
          <span
            className="button"
          >
            测试专区
            <span
              className="pt-icon-standard pt-icon-chevron-down pt-align-right"
            />
          </span>
        </Popover>
        <Popover
          arrowSize={30}
          className="button-bg"
          content={
            <Blueprint.Menu
              className="menu"
            >
              <Blueprint.MenuItem
                className="subMenu"
                disabled={false}
                shouldDismissPopover={true}
                submenuViewportMargin={Object {}}
                text="共通性分析统计功能"
                useSmartPositioning={true}
              >
                <Link
                  className="menuItem"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/qca/stk"
                >
                  <Blueprint.Button
                    className="pt-menu-item pt-minimal menu-item-button"
                  />
                </Link>
                <Link
                  className="menuItem"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/qca/stk"
                >
                  <Blueprint.Button
                    className="pt-menu-item pt-minimal menu-item-button"
                    text="机台共通性分析"
                  />
                </Link>
                <Link
                  className="menuItem"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/qca/stk"
                >
                  <Blueprint.Button
                    className="pt-menu-item pt-minimal menu-item-button"
                    text="原物料共通性分析"
                  />
                </Link>
                <Link
                  className="menuItem"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/qca/stk"
                >
                  <Blueprint.Button
                    className="pt-menu-item pt-minimal menu-item-button"
                    text="制程参数（EDC）比对"
                  />
                </Link>
                <Link
                  className="menuItem"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/qca/echarts"
                >
                  <Blueprint.Button
                    className="pt-menu-item pt-minimal menu-item-button"
                    text="Echarts制图试用"
                  />
                </Link>
              </Blueprint.MenuItem>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
              <Link
                className="menuItem"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/qca/"
              >
                <Blueprint.Button
                  className="pt-menu-item pt-minimal menu-item-button"
                />
              </Link>
            </Blueprint.Menu>
          }
          defaultIsOpen={false}
          hoverCloseDelay={300}
          hoverOpenDelay={150}
          inheritDarkTheme={true}
          inline={false}
          interactionKind={0}
          isDisabled={false}
          isModal={false}
          popoverClassName=""
          position={7}
          rootElementTag="span"
          transitionDuration={300}
          useSmartArrowPositioning={true}
          useSmartPositioning={false}
        >
          <span
            className="button"
          >
            工程专区
            <span
              className="pt-icon-standard pt-icon-chevron-down pt-align-right"
            />
          </span>
        </Popover>
      </section>
    </div>
    <div
      className="pt-navbar-group pt-align-right"
    >
      <span
        className="pt-navbar-divider"
      />
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <Link
          className="link"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/dRTM/currentalarm"
        >
          dRTM
        </Link>
      </div>
      <span
        className="pt-navbar-divider"
      />
      <button
        className="pt-button pt-minimal pt-icon-notifications"
      />
      <button
        className="pt-button pt-minimal pt-icon-more"
      />
      <span
        className="pt-navbar-divider"
      />
      <span
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Log out
      </span>
    </div>
  </nav>
  <SubNav
    location="QCA"
    route={
      Object {
        "path": "qca(/:path)",
      }
    }
  />
</div>
`;
